src/main/java/io/aiven/connect/elasticsearch/bulk/BulkProcessor.java

```diff
@@ -68,13 +68,14 @@ public class BulkProcessor<R, B> {
     // but may be part of synchronized(this) wait() conditions so need to notifyAll() on changes
     private volatile boolean stopRequested = false;
     private volatile boolean flushRequested = false;
-    private final AtomicReference<ConnectException> error = new AtomicReference<>();
+    private final AtomicReference<RuntimeException> error = new AtomicReference<>();
```

Comment:
I guess this is the main fix here, but by changing it back to `ConnectException`, tests are still passing. Could we add a test that validates how things should work when a RuntimeException is thrown?

Reply:
nope, this was a diff I've missed to rollback before asking the review :)
it was my understanding before, the problem its not that `ConnectException` its retried by kafka, the problem its that if `dropInvalidMessage` is set to true the `ConnectException` was catched even in the `.put()` method, instead it shoud catch only the convertion errors, I've rollbacked to the previous state :)

Reply:
Got it. As it's not apparent on first sight, should we make a comment related to this 
> the problem its that if dropInvalidMessage is set to true the ConnectException was catched even in the .put() method, 

in the changes applied to `ElasticsearchWriter#tryWriteRecord` to clarify the approach taken?